src/events/attributes.rs

```diff
@@ -437,12 +438,139 @@ impl<'a> Iterator for Attributes<'a> {
                 self.position = j;
                 return attr!(start_key..end_key, i..j);
             }
-            Some((i, _)) => err!(Error::UnquotedValue(i)),
+            Some((i, _)) => err!(AttrError::UnquotedValue(i)),
             None => return attr!(start_key..end_key),
         }
     }
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Errors that can be raised during parsing attributes.
+///
+/// Recovery position in examples shows the position from which parsing of the
+/// next attribute will be attempted.
+#[derive(Debug, PartialEq)]
+pub enum AttrError {
+    /// Attribute key not followed by with `=`, position relative to start of
+    /// owning tag is provided.
+    ///
+    /// Example of input that raises this error:
+    ///
+    /// ```xml
+    /// <tag key another="attribute"/>
+    /// <!--     ^~~ error position, recovery position (8) -->
+    /// ```
+    ///
+    /// That error can be raised only if iterator in XML mode.
+    ExpectedEq(usize),
+    /// Attribute value not found after `=`, position relative to start of owning
+    /// tag is provided.
+    ///
+    /// Example of input that raises this error:
+    ///
+    /// ```xml
+    /// <tag key = />
+    /// <!--       ^~~ error position, recovery position (10) -->
+    /// ```
+    ///
+    /// Notice, that this error can be returned only for the last attribute in
```

Comment:
```suggestion
    /// This error can be returned only for the last attribute in
```

Removing the "notice" here and below is a very minor suggestion, again, feel free to ignore it.  It just reads better without it in my opinion.

Reply:
I will rely on your opinion :)